osquery/logger/logger.cpp

```diff
@@ -90,9 +90,25 @@ class BufferedLogSink : public google::LogSink, private boost::noncopyable {
 
   /// Set the forwarding mode of the buffering sink.
   static void forward(bool forward = false) {
+    WriteLock lock(instance().forward_mutex_);
```

Comment:
It maybe overly defensive, but the idea was to use the same mutex as used to disable/restore forwarding, so one thread can't call forward(true) while another thinks it has disabled and locked the logger.